File: 设计模式/程序/24-VisitorPattern/VisitorPattern(C#)/ComputerPart.cs
using System;

namespace VisitorPatternApp
{
    /* 电脑部件基类 */  
    public interface ComputerPart
    {
        void Accept(ComputerPartVisitor visitor);
    }

    /* 电脑部件实现 */
    public class Keyboard : ComputerPart
    {
        public void Accept(ComputerPartVisitor visitor)
        {
            visitor.Visit(this);
        }
    }

    public class Monitor : ComputerPart
    {
        public void Accept(ComputerPartVisitor visitor)
        {
            visitor.Visit(this);
        }
    }

    public class Mouse : ComputerPart
    {
        public void Accept(ComputerPartVisitor visitor)
        {
            visitor.Visit(this);
        }
    }
}

File: 设计模式/程序/24-VisitorPattern/VisitorPattern(python)/ComputerPart.py
import abc

# 电脑部件基类
class ComputerPart(metaclass = abc.ABCMeta):
    @abc.abstractmethod
    def Accept(self, visitor):
        pass

# 电脑部件子类
class Keyboard(ComputerPart):
    def Accept(self, visitor):
        visitor.VisitKeyborad(self)

class Monitor(ComputerPart):
    def Accept(self, visitor):
        visitor.VisitMonitor(self)

class Mouse(ComputerPart):
    def Accept(self, visitor):
        visitor.VisitMouse(self)
